### PR TITLE
Update SIN-4-FP-21 and SIN-4-FP-20 from NodOn

### PR DESCRIPTION
**Pull request by AlexisPolegato**
Update SIN-4-FP-21 from NodOn to integrate Fil Pilote Cluster
Add SIN-4-FP-20 from NodOn

### CODE REVIEW

File: src/devices/nodon.ts [hunk 1]
```diff
@@ -114,22 +162,52 @@ const definitions: Definition[] = [
...
-        fromZigbee: [fz.on_off, fz.metering],
-        toZigbee: [tz.on_off],
-        exposes: [e.switch(), e.power(), e.energy()],
+        fromZigbee: [fz.on_off, fz.metering, fzLocal.fil_pilote_mode],
+        toZigbee: [tz.on_off, tzLocal.fil_pilote_mode],
+        exposes: [
+            e.switch(),
+            e.power(),
+            e.energy(),
+            e.enum('mode', ea.ALL, ['Comfort', 'Eco', 'Anti-Freeze', 'Stop', 'Comfort -1', 'Comfort -2']),
```
Comment (Koenkk) [inline src/devices/nodon.ts:201]: Please make these snake case, e.g. `anti-freeze`, `comfort_-1`

**Comment by krikristoophe** [comment]
Hello @AlexisPolegato @Koenkk 

This PR was made while I create one for `SIN-4-FP-21_EQU` [here](https://github.com/Koenkk/zigbee-herdsman-converters/pull/6434). As @yoan-m [said](https://github.com/Koenkk/zigbee2mqtt/issues/19169#issuecomment-1801226560), it seem that  `SIN-4-FP-21_EQU` work same as `SIN-4-FP-21`. Maybe we can add `SIN-4-FP-21_EQU` to model list in this PR ?

Also, theses models works like `Legrand 064882` ([Device definition](https://github.com/Koenkk/zigbee-herdsman-converters/blob/3afd0be2a6dcc5f863d7499923b5884efb02c61a/src/devices/legrand.ts#L530)), maybe it can be good to unify values like :
- `mode` -> `cable_outlet_mode `
-  `e.enum('mode', ea.ALL, ['comfort', 'eco', 'anti-freeze', 'stop', 'comfort_-1', 'comfort_-2']),` -> `e.enum('cable_outlet_mode', ea.ALL, ['comfort', 'comfort-1', 'comfort-2', 'eco', 'frost_protection', 'off']),`

It's not a big change and can avoid that all future similar devices use it own values no ?

File: src/devices/nodon.ts [hunk 2]
```diff
@@ -1,11 +1,59 @@
...
 import * as ota from '../lib/ota';
+import * as utils from '../lib/utils';
 const e = exposes.presets;
+const ea = exposes.access;
 import tz from '../converters/toZigbee';
 import fz from '../converters/fromZigbee';
+const manufacturerOptions = {manufacturerCode: 0x128B};
+
+const fzLocal = {
+    fil_pilote_mode: {
```
Comment (Koenkk) [inline src/devices/nodon.ts:14]: Since this code is duplicated, could you move it to `fromZigbee.ts`/`toZigbee.ts`?

**Comment by Koenkk** [comment]
Thanks!

File: src/converters/fromZigbee.ts [hunk 3]
```diff
@@ -6323,6 +6323,26 @@ const converters2 = {
             }
         },
     } as Fz.Converter,
+    nodon_fil_pilote_mode: {
+        cluster: 'manuSpecificNodOnFilPilote',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const payload: KeyValueAny = {};
+            const mode = msg.data['0'];
```
Comment (yoan-m) [inline src/converters/fromZigbee.ts:6331]: Tt shouldn't be `const mode = msg.data['mode'];` ?

Comment (krikristoophe) [reply src/converters/fromZigbee.ts:6331]: Yes you're right ! I seen it this morning I didn't had time to update my PR to fix it

Comment (yoan-m) [reply src/converters/fromZigbee.ts:6331]: Done here [https://github.com/Koenkk/zigbee-herdsman-converters/pull/6465](https://github.com/Koenkk/zigbee-herdsman-converters/pull/6465)

**Comment by PoloAlexis** [comment]
Hello all

I found some issue with the oscilloscope
Please review
https://github.com/Koenkk/zigbee-herdsman-converters/pull/6473

**Pull request merged**